### PR TITLE
feat(docs): Add Sourcing from Sanity Documentation

### PR DESCRIPTION
**Pull request by kmelve**
## Description

This PR adds documentation for how to use Sanity.io as a content backend for Gatsby.

### CODE REVIEW

**Comment by KyleAMathews** [comment]
This needs linked to from the docs-links.yaml nav.

**Comment by KyleAMathews** [comment]
So it shows up on .org

File: docs/docs/sourcing-from-sanity.md [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+You can also check out [the company website example](https://github.com/sanity-io/example-company-website-gatsby-sanity-combo) we have set up. It contains both a configured Sanity Studio and a Gatsby frontend, which you can get up and running within minutes. It can be an useful reference for how to build a website using structured content. Follow the instructions in its README.md to get up and running.
+
+This guide will cover how configure and use the [`gatsby-source-sanity`](https://www.npmjs.com/package/gatsby-source-sanity) plugin.
+
+## Basic usage
+
+```
+yarn add gatsby-source-sanity
+# or
+npm i gatsby-source-sanity --save
```
[the remark this answers is not in the document]
Comment (kmelve) [reply docs/docs/sourcing-from-sanity.md]: Good point! Fixed in new commit 👍

File: docs/docs/sourcing-from-sanity.md [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+```
+
+```js
+// in your gatsby-config.js
+module.exports = {
+  // ...
```
[the remark this answers is not in the document]
Comment (kmelve) [reply docs/docs/sourcing-from-sanity.md]: Done!

File: docs/docs/sourcing-from-sanity.md [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+  plugins: [
+    {
+      resolve: "gatsby-source-sanity",
+      options: {
+        projectId: "abc123",
+        dataset: "blog",
+        token: process.env.MY_SANITY_TOKEN || "my-token",
```
[the remark this answers is not in the document]
Comment (kmelve) [reply docs/docs/sourcing-from-sanity.md]: Agree – fixed in new commit 👍

**Comment by calcsam** [comment]
Let's merge this in, since it is ready. @kmelve you should add a link to this doc here: https://www.gatsbyjs.org/docs/headless-cms/

**Pull request merged**